pkg/api/types.go

```diff
@@ -450,10 +452,81 @@ type StepConfiguration struct {
 // if no explicit output tag is provided, the name
 // of the image is used as the tag.
 type InputImageTagStepConfiguration struct {
+	InputImage `json:",inline"`
+	Sources    *[]ImageStreamSource `json:"-"`
+}
+
+func (config InputImageTagStepConfiguration) Matches(other InputImage) bool {
+	return config.InputImage == other
+}
+
+func (config InputImageTagStepConfiguration) FormattedSources() string {
+	var formattedSources string
+	tests := sets.String{}
+	if config.Sources != nil && len(*config.Sources) > 0 {
+		for _, source := range *config.Sources {
+			switch source.SourceType {
+			case ImageStreamSourceTest:
+				tests.Insert(source.Name)
+			default:
+				formattedSources += fmt.Sprintf("%s,", string(source.SourceType))
+			}
+		}
+
+		if len(tests) > 0 {
+			var testNames []string
+			for k := range tests {
+				testNames = append(testNames, k)
+			}
+			sort.Strings(testNames)
+			formattedSources += fmt.Sprintf("test steps: %s", strings.Join(testNames, ","))
+		}
+
+		return strings.TrimSuffix(formattedSources, ",")
```

Comment:
nit: you could also keep `formattedSources` as a slice and return `strings.Join(formattedSources, ",")`